R/as_sass.R

```diff
@@ -81,6 +81,22 @@ find_dependencies <- function(x) {
   c(childDeps, deps)
 }
 
+discard_dependencies <- function(x) {
+  if (is_sass_bundle_like(x))  {
+    x <- as_sass_layer(x)
+    x$html_deps <- NULL
+  } else if (is_font_collection(x)) {
+    x$html_deps <- NULL
+  } else {
+    htmlDependencies(x) <- NULL
```

Comment:
Note that this won't remove an html dependency if it's included as a child of a tag object (as opposed to being added with `attachDependency()`):

```R
x <- div(
  "hello",
  htmlDependency("x", "1.0", c(src = "abc"))
)
htmlDependencies(x) <- NULL
findDependencies(x)
#> [[1]]
#> List of 10
#>  $ name      : chr "x"
#>  $ version   : chr "1.0"
#>  $ src       :List of 1
#>   ..$ src: chr "abc"
#>  $ meta      : NULL
#>  $ script    : NULL
#>  $ stylesheet: NULL
#>  $ head      : NULL
#>  $ attachment: NULL
#>  $ package   : NULL
#>  $ all_files : logi TRUE
#>  - attr(*, "class")= chr "html_dependency"
```

Same is true if it's an element in a `tagList` or `list`:

```R
x <- tagList(
  div("hello"),
  htmlDependency("x", "1.0", c(src = "abc"))
)
htmlDependencies(x) <- NULL
findDependencies(x)
#> [[1]]
#> List of 10
#>  $ name      : chr "x"
#>  $ version   : chr "1.0"
#>  $ src       :List of 1
#>   ..$ src: chr "abc"
#>  $ meta      : NULL
#>  $ script    : NULL
#>  $ stylesheet: NULL
#>  $ head      : NULL
#>  $ attachment: NULL
#>  $ package   : NULL
#>  $ all_files : logi TRUE
#>  - attr(*, "class")= chr "html_dependency"
```

Reply:
I don't think this is an actual issue since `as_sass()` currently doesn't have a method for tags for dependencies (this is partially the motivation for having the `sass_layer(html_deps)`):

```r
> sass::as_sass(list(".foo{}", htmltools::htmlDependency("foo", "1.0", ".")))
Error in as_sass_.default(val) : 
  as_sass() not implemented for object of class: 'html_dependency'
```

That being said, maybe it still make sense to add a case:

```r
} else if (inherits(x, "html_dependency")) {
    x <- NULL
  }
```

just in case we decide to add a no-op method someday